xs2a-adapter-service-api-adapter/src/main/java/de/adorsys/xs2a/adapter/adapter/BaseAccountInformationService.java

```diff
@@ -258,17 +260,25 @@ private String getTransactionListUri(String accountId, RequestParams requestPara
             .headers(headersMap)
             .send(requestBuilderInterceptor, jsonResponseHandler(klass));
         TransactionsReport transactionsReport = mapper.apply(response.getBody());
-
+        logTransactionsSize(transactionsReport);
         return new Response<>(response.getStatusCode(), transactionsReport, response.getHeaders());
     }
 
+    private void logTransactionsSize(TransactionsReport transactionsReport) {
+        AccountReport transactions = transactionsReport.getTransactions();
+        int size = transactions.getBooked().size() + transactions.getPending().size();
```

Comment:
I am afraid, that if PSU has no pending transactions, `transactions.getPending()` might return `null`.
The same for `transactions.getBooked()`.

Am I wrong?

Reply:
you're totally right, fixed